cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -4927,6 +4927,160 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
+                            let pxor_const =
+                                ctx.use_constant(VCodeConstantData::WellKnown(&PXOR_CONST));
+                            let pxor_const_reg = ctx.alloc_tmp(types::I16X8).only_reg().unwrap();
+                            ctx.emit(Inst::xmm_load_const(
+                                pxor_const,
+                                pxor_const_reg,
+                                types::I16X8,
+                            ));
+                            ctx.emit(Inst::xmm_mov(SseOpcode::Movdqa, RegMem::reg(src), dst));
+                            ctx.emit(Inst::xmm_rm_r(
+                                SseOpcode::Pxor,
+                                RegMem::reg(pxor_const_reg.to_reg()),
+                                dst,
+                            ));
+
+                            static MADD_CONST: [u8; 16] = [
+                                0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01,
+                                0x00, 0x01, 0x00, 0x01, 0x00,
+                            ];
+                            let madd_const =
+                                ctx.use_constant(VCodeConstantData::WellKnown(&MADD_CONST));
+                            let madd_const_reg = ctx.alloc_tmp(types::I8X16).only_reg().unwrap();
+                            ctx.emit(Inst::xmm_load_const(
+                                madd_const,
+                                madd_const_reg,
+                                types::I16X8,
+                            ));
+                            ctx.emit(Inst::xmm_rm_r(
+                                SseOpcode::Pmaddwd,
+                                RegMem::reg(madd_const_reg.to_reg()),
+                                dst,
+                            ));
+                            static ADDD_CONST2: [u8; 16] = [
+                                0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
+                                0x00, 0x00, 0x00, 0x01, 0x00,
+                            ];
+                            let addd_const2 =
+                                ctx.use_constant(VCodeConstantData::WellKnown(&ADDD_CONST2));
+                            let addd_const2_reg = ctx.alloc_tmp(types::I8X16).only_reg().unwrap();
+                            ctx.emit(Inst::xmm_load_const(
+                                addd_const2,
+                                addd_const2_reg,
+                                types::I16X8,
+                            ));
+                            ctx.emit(Inst::xmm_rm_r(
+                                SseOpcode::Paddd,
+                                RegMem::reg(addd_const2_reg.to_reg()),
+                                dst,
+                            ));
+                        }
+                        _ => {
+                            unimplemented! {"Type not supported for {:?}", op}
+                        }
+                    }
+                }
+            }
```

Comment:
I think we need an `else` branch here to handle a pairwise add without the extends, no? Or at least an `unimplemented!()` if we don't support that opcode by itself.